util/memory/tracker.go

```diff
@@ -223,6 +226,29 @@ func (t *Tracker) Consume(bytes int64) {
 	}
 }
 
+// consumeNegative takes a negative value which means this is a memory release operation.
+// It does the same thing as Consume without checking exceeding,
+// because a negative value shouldn't trigger Exceed Action.
+// It will panic if "bytes" is positive.
+func (t *Tracker) consumeNegative(bytes int64) {
```

Comment:
What about move `consumeNegative` into `Consume` function, then if `Consume` receiving negative value, it would delegate to the `consumeNegative` the function.

Reply:
Good idea.